src/main/java/com/indix/client/impl/IndixApiClientImpl.java

```diff
@@ -245,6 +244,7 @@ public OffersSearchResult getProductsOffersPremium(Query query) throws IndixApiE
                     });
             return searchIndixApiResponse.getResult();
         } catch (IndixApiException iae) {
+            logger.error("getProductsOffersPremium failed: " + iae.getMessage());
             throw iae;
         } catch (Exception e) {
             logger.error("getProductsOffersPremium failed: " + e.getMessage());
```

Comment:
Ashwanth, I agree with your point. Shreya, lets discuss on this tomorrow and take a call.